app/controllers/api/migration_analytics_manifest_controller.rb

```diff
@@ -0,0 +1,34 @@
+def parse_json_if_valid(json)
```

Comment:
This should be a private method in your controller class

Reply:
That makes sense. I'm too used to file-scoped functions in my JS :)